### ISSUE TITLE
URL in the sitemap is not encoded correctly

### ISSUE THREAD

**Issue by newtonex** [boazpoolman/strapi-plugin-sitemap#110]
### Describe the bug
URL is not encoded correctly. Hence, broken links are created in the sitemap.

### Steps to reproduce the behavior
Example URL
https://example.com/Can%20You%20Easily%20Build%20a%20Sitemap%3F

Result in sitemap
https://example.com/Can%20You%20Easily%20Build%20a%20Sitemap?

### Expected behavior
Allowed field should always be encoded as URL component.

### Additional context
I did read a bit in the code to see if there is any work around without changing the core. I can't find encodeURIComponent anywhere.

**Comment by boazpoolman** [comment]
I don't quite understand what is the problem here.
Could you rephrase?

**Comment by Can-G-Zaleo** [comment]
I think i understand because i had the same problem.

This plugin provides placeholder in the Pattern Input.
![image](https://github.com/boazpoolman/strapi-plugin-sitemap/assets/66360899/a15d5d13-57ba-4a38-ae73-342b00286863)
When the placeholder is a custom field from the collection-type and incude spaces. It translates the spaces to "%20" by generating the URL. It would be cool if you could catch this special case in the plugin by being able to decide for yourself which character to replace the spaces with.

My current workaround is to get the sitemap data as a data source and use a regex to replace the spaces (%20) with a minus (-) and use the revised xml as the sitemap actual sitemap.xml

I hope I could help :)

**Comment by boazpoolman** [comment]
Gotcha.

That's why (by default) I only allow you to use fields of type `id` and `uid`.

If you go and alter the settings to also allow `string` or `text` fields the plugin can't be held accountable for spaces in your URLs.

Don't get me wrong, your solution of replacing the spaces with dashes is very viable.
Though what bothers me is that the URL as stated in the sitemap should actually be the same as the URL as it is composed in your frontend.

As soon as we start rewriting the URL in the backend with some magic rules, those magic rules need to be applied in the frontend as well to make sure the URLs are the same. 
That is just prone to errors if you ask me.

I strive to merge this plugin with the [URL alias plugin](https://github.com/strapi-community/strapi-plugin-url-alias) which allready shares a lot of logic with the sitemap plugin. I wan't to do this in a way so that you have to set the URL once for both the sitemap generation and front-end routing.

Anywho, I'm rambling. If you wanna help with the above HMU.
I won't be rewriting URLs in this plugin though.

**Comment by boazpoolman** [comment]
Closing as this won't be worked on.
